### PR TITLE
Exclude jsdom when bundling for headless tests

### PR DESCRIPTION
**Pull request by mgred**
#### Purpose (TL;DR) - mandatory

Use Option `--no-detect-globals` for `mochify` command in `test-headless`

#### Background (Problem in detail)  - optional

Based on the suggestion by @mantoni [here](https://github.com/sinonjs/samsam/pull/130#issuecomment-541600189)

#### How to verify - mandatory
1. Check out this branch
2. `npm install`
3. `npm run test`, `npm run test-headless`

#### Checklist for author

- [x] `npm run lint` passes
- [x] References to standard library functions are [cached](https://github.com/sinonjs/sinon/pull/1523).

### CODE REVIEW

File: lib/is-element.test.js [hunk 1]
```diff
@@ -2,24 +2,24 @@
...
-    if (require.resolve) {
-        delete require.cache[require.resolve(file)];
-    }
-
-    return require("./is-element");
-}
-
 describe("isElement", function() {
-    var isElement;
+    var isElement = require("./is-element");
```
Comment (mroderick) [inline lib/is-element.test.js]: I don't think the require statement here does anything we need?

Comment (mgred) [reply lib/is-element.test.js]: That's right. I will remove it, thanks 👌

**Comment by mgred** [comment]
@mroderick I removed the unnecessary `require` call

**Comment by mgred** [comment]
> Looks good to me.
> 
> Could you rebase it so the last commit goes away?

I do

**Pull request merged**